applications/tari_base_node/src/command_handler.rs

```diff
@@ -970,31 +970,29 @@ impl CommandHandler {
                 let header = try_or_print!(db.fetch_chain_header(height).await);
 
                 // Optionally, filter out pow algos
-                if pow_algo.map(|algo| header.header.pow_algo() != algo).unwrap_or(false) {
+                if pow_algo.map(|algo| header.header().pow_algo() != algo).unwrap_or(false) {
                     continue;
                 }
 
                 let target_diff = try_or_print!(db.fetch_target_difficulties(prev_header.hash().clone()).await);
+                let pow_algo = header.header().pow_algo();
 
-                let min = consensus_rules
-                    .consensus_constants(height)
-                    .min_pow_difficulty(header.header.pow_algo());
-                let max = consensus_rules
-                    .consensus_constants(height)
-                    .max_pow_difficulty(header.header.pow_algo());
-
-                let calculated_target_difficulty = target_diff.get(header.header.pow_algo()).calculate(min, max);
-                let existing_target_difficulty = header.accumulated_data.target_difficulty;
-                let achieved = header.accumulated_data.achieved_difficulty;
-                let solve_time = header.header.timestamp.as_u64() as i64 - prev_header.header.timestamp.as_u64() as i64;
+                let min = consensus_rules.consensus_constants(height).min_pow_difficulty(pow_algo);
+                let max = consensus_rules.consensus_constants(height).max_pow_difficulty(pow_algo);
+
+                let calculated_target_difficulty = target_diff.get(pow_algo).calculate(min, max);
+                let existing_target_difficulty = header.accumulated_data().target_difficulty;
+                let achieved = header.accumulated_data().achieved_difficulty;
+                let solve_time =
+                    header.header().timestamp.as_u64() as i64 - prev_header.header().timestamp.as_u64() as i64;
                 let normalized_solve_time = cmp::min(
                     cmp::max(solve_time, 1) as u64,
                     consensus_rules
                         .consensus_constants(height)
-                        .get_difficulty_max_block_interval(header.header.pow_algo()),
+                        .get_difficulty_max_block_interval(pow_algo),
                 );
-                let acc_sha3 = header.accumulated_data.accumulated_blake_difficulty;
-                let acc_monero = header.accumulated_data.accumulated_monero_difficulty;
+                let acc_sha3 = header.accumulated_data().accumulated_blake_difficulty;
```

Comment:
`accumulated_blake_difficulty`
We should really put in a todo to refactor this next reset.

Reply:
Yeah agree - I'll add a "tagged comment" #testnetreset on the field. I suspect we could rename it now without breaking any serialised data, but havent tried.